rasa/utils/tensorflow/tf_models.py

```diff
@@ -59,20 +58,24 @@ def train_dataset_function(_batch_size):
 
         def evaluation_dataset_function(_batch_size):
             return evaluation_model_data.as_tf_dataset(
-                _batch_size, batch_strategy, shuffle=False
+                _batch_size, "sequence", shuffle=False
             )
 
         if eager:
             # allows increasing batch size
             tf_train_dataset_function = train_dataset_function
             tf_train_on_batch_function = self.train_on_batch
         else:
+            logger.debug("Building tensorflow train graph...")
             # allows increasing batch size
             tf_train_dataset_function = tf.function(func=train_dataset_function)
+            init_dataset = tf_train_dataset_function(tf_batch_size)
```

Comment:
Why did you add this?

Reply:
to prebuild tf graph before tqdm starts, separate building and running time. it is a part of the next PR

Reply:
ok, great. thanks for clarifying.